flightControl/src/AvState.cpp

```diff
@@ -0,0 +1,262 @@
+#include "AvState.h"
+#include <string.h>
+#include <list>
+#include <sensors.h>
+#include "telecom.h"
+#include "PacketDefinition.h"
+
+
+AvState::AvState(const Thresholds& thresholds) : thresholds(thresholds)
+{
+    this->currentState = State::IDLE;
+}
+
+// destructor
+AvState::~AvState()
```

Comment:
This makes no sense. Since the data WILL be destroyed, and the address be made out-of-bounds for the program, why force a value onto it ?